src/test/java/com/n3twork/dynamap/DynamapTest.java

```diff
@@ -794,6 +794,50 @@ public void testDelete() {
         Assert.assertNull(doc);
     }
 
+
+    @Test
+    public void testDeleteWithCondition() {
+
+        TestDocumentBean doc = createTestDocumentBean(createNestedTypeBean());
+        dynamap.save(new SaveParams<>(doc));
+
+        doc = dynamap.getObject(createGetObjectParams(doc));
+        Assert.assertNotNull(doc);
+
+        WriteConditionCheck<TestDocumentBean> failCheck = new WriteConditionCheck<>(TestDocumentBean.class, doc.getId(), null);
+        DynamoExpressionBuilder failBuilder = failCheck.getDynamoExpressionBuilder();
+        failBuilder.addAttributeExistsCondition(String.join("", UUID.randomUUID().toString().split("-")));  // check for an attribute that doesn't exist to purposefully fail
+
+        DeleteRequest<TestDocumentBean> failDeleteRequest = new DeleteRequest<>(TestDocumentBean.class)
+                .withHashKeyValue(doc.getId())
+                .withRangeKeyValue(doc.getRangeKeyValue())
+                .withConditionExpression(failBuilder.buildConditionalExpression());
+
+        try {
+            dynamap.delete(failDeleteRequest);
```

Comment:
Suggestion: Could use something like `assertThrows()` to keep simple

Reply:
Good point. I looked and it doesn't seem like that is available in this version of testng, sadly.

Reply:
Testng takes an expected exceptions parameter in its test annotation 
So you could split this into two tests and use that parameter